test/compilable/test17502.d

```diff
@@ -0,0 +1,30 @@
+class Foo
+{
+    auto foo()
+    out {}
+    body {}
+
+    auto bar()
+    out { assert (__result > 5); }
```

Comment:
I'm not sure using that private identifier is something we want to advocate / test :D